### PR TITLE
initial pass at config plugin arguments refactor

### PR DESCRIPTION
**Pull request by michaelshobbs**
- adds `--global` for managing `$DOKKU_ROOT/ENV`
- adds `[config:set|config:unset] --no-restart`
- deprecates `config:set-norestart` and `config:unset-norestart`

```
root@dokku:~/dokku# dokku config --global
=====> global config vars
CURL_CONNECT_TIMEOUT: 5
CURL_TIMEOUT:         30
```

```
root@dokku:~/dokku# dokku config:set --no-restart node-js-app testing=true
-----> Setting config vars
       testing: true
```

### CODE REVIEW

**Comment by michaelshobbs** [comment]
refs: #1261

File: plugins/backup/commands [hunk 1]
```diff
@@ -87,8 +87,8 @@ case "$1" in
 
   help | backup:help)
     cat && cat<<EOF
-    backup:export [file]                            Export dokku configuration files
-    backup:import [file]                            Import dokku configuration files
+    backup:export [file]                                       Export dokku configuration files
+    backup:import [file]                                       Import dokku configuration files
```
Comment (josegonzalez) [inline plugins/backup/commands]: Lol when are these rando-spaces going away?

File: plugins/config/commands [hunk 2]
```diff
@@ -46,14 +48,36 @@ config_write() {
...
+  case "$2" in
+    --global)
+      ENV_FILE="$DOKKU_ROOT/ENV"
+      DOKKU_CONFIG_TYPE="global"
+      DOKKU_CONFIG_RESTART=false
+      ;;
+    --no-restart)
+      ARGS=("$@") && shift 2
+      DOKKU_CONFIG_RESTART=false dokku ${ARGS[0]} "$@"
+      exit 0
```
Comment (josegonzalez) [inline plugins/config/commands]: Shouldn't this exit the status of the last call?

File: plugins/backup/commands (hunk 1 above)
Comment (michaelshobbs) [inline plugins/backup/commands]: When you fix them, duh! :smiling_imp:

File: plugins/config/commands (hunk 2 above)
Comment (michaelshobbs) [inline plugins/config/commands]: Should? Yes. I'm not satisfied with this. Let me take another pass at it.

File: plugins/backup/commands (hunk 1 above)
Comment (josegonzalez) [inline plugins/backup/commands]: :D

File: plugins/config/commands [hunk 3]
```diff
@@ -24,7 +22,7 @@ config_styled_hash () {
 
   while read -r word; do
     KEY=$(echo $word | cut -d"=" -f1)
-    VALUE=$(echo $word | cut -d"=" -f2- | sed -e "s/^'//" -e "s/'$//")
+    VALUE=$(echo $word | cut -d"=" -f2- | sed -e "s/^'//" -e "s/'$//" -e "s/\$$//g")
```
Comment (michaelshobbs) [inline plugins/config/commands]: strips trailing `$` from output of `read -r`

File: plugins/config/commands [hunk 4]
```diff
@@ -46,14 +46,43 @@ config_write() {
...
+      ENV_FILE="$DOKKU_ROOT/ENV"
+      DOKKU_CONFIG_TYPE="global"
+      DOKKU_CONFIG_RESTART=false
+      ;;
+    --no-restart)
+      APP="$3"
+      ENV_FILE="$DOKKU_ROOT/$APP/ENV"
+      DOKKU_CONFIG_RESTART=false
+      DOKKU_CONFIG_TYPE="app"
+      set -- "${@:1:1}" "${@:3}"
```
Comment (michaelshobbs) [inline plugins/config/commands]: removes `$2` from `$@`

File: plugins/config/commands [hunk 5]
```diff
@@ -90,26 +118,8 @@ case "$1" in
...
-    dokku_log_info1 "Restarting app $APP"
-    dokku ps:restart $APP
-   ;;
-
-  config:set-norestart)
-    [[ -z $2 ]] && echo "Please specify an app to run the command on" && exit 1
-    verify_app_name "$2"
-    APP="$2"
+    parse_config_args "$@"
+    [[ "$2" = "--no-restart" ]] && set -- "${@:1:1}" "${@:3}"
```
Comment (michaelshobbs) [inline plugins/config/commands]: couldn't get around doing this again as `parse_config_args` is run in subshell, unless we have two functions (one to set env vars and one to echo new `$@` and use it. meh...

**Comment by michaelshobbs** [comment]
blocks #1261

**Comment by josegonzalez** [comment]
Looks good to me. Merge when ready :)

**Pull request merged**